File: fix_remaining_quotes.py
#!/usr/bin/env python3
"""
Script to fix remaining quote entities in React files.
"""

import os
import re
from pathlib import Path

def fix_remaining_quotes(file_path):
    """Fix remaining quote entities in a file."""
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Fix any remaining &quot; entities
    content = re.sub(r'&quot;', '"', content)
    
    # Write the fixed content
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    return True

def main():
    """Main function to fix all remaining quote entities."""
    
    # Find all React component files
    component_files = []
    
    # Search in app directory
    app_dir = Path('/workspace/app')
    if app_dir.exists():
        component_files.extend(app_dir.rglob('*.tsx'))
    
    # Search in src directory
    src_dir = Path('/workspace/src')
    if src_dir.exists():
        component_files.extend(src_dir.rglob('*.tsx'))
    
    fixed_count = 0
    
    for file_path in component_files:
        try:
            if fix_remaining_quotes(str(file_path)):
                fixed_count += 1
                print(f"Fixed remaining quotes in: {file_path}")
        except Exception as e:
            print(f"Error processing {file_path}: {e}")
    
    print(f"\nFixed remaining quotes in {fixed_count} files")

if __name__ == '__main__':
    main()

File: fix-remaining-quotes.js
const fs = require('fs');
const path = require('path');

// Function to fix remaining quote issues
function fixRemainingQuotes(content) {
  // Fix mixed quotes in object properties
  content = content.replace(/name: "([^"]+)&quot;/g, 'name: "$1"');
  content = content.replace(/path: "([^"]+)&quot;/g, 'path: "$1"');
  content = content.replace(/title: "([^"]+)&quot;/g, 'title: "$1"');
  content = content.replace(/description: "([^"]+)&quot;/g, 'description: "$1"');
  content = content.replace(/className: "([^"]+)&quot;/g, 'className: "$1"');
  content = content.replace(/href: "([^"]+)&quot;/g, 'href: "$1"');
  
  // Fix mixed quotes in JSX attributes
  content = content.replace(/className="([^"]+)&quot;/g, 'className="$1"');
  content = content.replace(/href="([^"]+)&quot;/g, 'href="$1"');
  content = content.replace(/src="([^"]+)&quot;/g, 'src="$1"');
  content = content.replace(/alt="([^"]+)&quot;/g, 'alt="$1"');
  content = content.replace(/type="([^"]+)&quot;/g, 'type="$1"');
  content = content.replace(/id="([^"]+)&quot;/g, 'id="$1"');
  
  // Fix any remaining &quot; patterns
  content = content.replace(/&quot;/g, '"');
  content = content.replace(/&apos;/g, "'");
  
  return content;
}

// Function to process a file
function processFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    const fixedContent = fixRemainingQuotes(content);
    
    if (content !== fixedContent) {
      fs.writeFileSync(filePath, fixedContent, 'utf8');
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Function to recursively find and process files
function processDirectory(dirPath) {
  let fixedCount = 0;
  
  try {
    const items = fs.readdirSync(dirPath);
    
    for (const item of items) {
      const fullPath = path.join(dirPath, item);
      const stat = fs.statSync(fullPath);
      
      if (stat.isDirectory() && !item.startsWith('.') && item !== 'node_modules') {
        fixedCount += processDirectory(fullPath);
      } else if (stat.isFile() && (item.endsWith('.tsx') || item.endsWith('.ts'))) {
        if (processFile(fullPath)) {
          fixedCount++;
        }
      }
    }
  } catch (error) {
    console.error(`Error processing directory ${dirPath}:`, error.message);
  }
  
  return fixedCount;
}

// Main execution
console.log('Starting remaining quote fixes...');
const fixedCount = processDirectory('./app');
const fixedCountSrc = processDirectory('./src');
console.log(`Fixed ${fixedCount + fixedCountSrc} files`);